test/runnable/test16640.d

```diff
@@ -0,0 +1,11 @@
+// PERMUTE_ARGS:
+
+void testFileFullPathAsDefaultArgument(string preBakedFileFullPath, string fileFullPath = __FILE_FULL_PATH__)
+{
+    assert(preBakedFileFullPath == fileFullPath);
+}
+
+void main()
+{
+    testFileFullPathAsDefaultArgument(__FILE_FULL_PATH__);
```

Comment:
Hmm imho you should inject the "real" path from the outside call `testFileFullPathAsDefaultArgument` as well - otherwise this test might be completely worthless if __FILE_FULL_PATH__ is broken.